unit_tests/testStrings/testStrings.f90

```diff
@@ -14,408 +14,475 @@ PROGRAM testStrings
+    ASSERT_EQ(CHAR(s2a(2,1)),"test2","rray")
+    ASSERT_EQ(CHAR(s2a(1,1)),"test2","rray")
+    ASSERT_EQ(CHAR(s2a(2,2)),"test2","rray")
+    DEALLOCATE(s2a2)
+
+    ALLOCATE(s2a2(2,3))
+    s2a2='one'
+    s2a=s2a2(:,1:2)
+    ASSERT(ALLOCATED(s2a),'allocated array assignment')
+    ASSERT(SIZE(s2a,DIM=1) == 2 .AND. SIZE(s2a,DIM=2) == 2,'SIZE array assignment')
+    ASSERT_EQ(CHAR(s2a(1,1)),"one","rray")
+    ASSERT_EQ(CHAR(s2a(2,1)),"one","rray")
+    ASSERT_EQ(CHAR(s2a(1,1)),"one","rray")
+    ASSERT_EQ(CHAR(s2a(2,2)),"one","rray")
+    DEALLOCATE(s2a)
+    DEALLOCATE(s2a2)
+   
+  ENDSUBROUTINE testAssign_arrays
 !
-!Test ALL operator
-  !Test ALL of 1-D array and scalar
-  COMPONENT_TEST('ALL 1-D & Scalar')
-  DO i=1,10
-    testStringArray(i)='test'
-  ENDDO
-  testString='bad test'
-  char10='bad test'
-  !ASSERT(.NOT. ALL(testString,testStringArray),'ALL(testString,testStringArray), bad')
-  ASSERT(.NOT. ALL(testStringArray,testString),'ALL 1-D & scalar, str')
-  !ASSERT(.NOT. ALL(char10,testStringArray),'ALL(char10,testStringArray), bad')
-  ASSERT(.NOT. ALL(testStringArray,char10),'ALL 1-D & scalar, char')
-  !ASSERT(ALL(testString,testStringArray,.TRUE.),'ALL(testString,testStringArray,.TRUE.), bad')
-  ASSERT(ALL(testStringArray,testString,.TRUE.),'ALL 1-D & scalar, str, w/ NOT=T')
-  !ASSERT(ALL(char10,testStringArray,.TRUE.),'ALL(char10,testStringArray,.TRUE.), bad')
-  ASSERT(ALL(testStringArray,char10,.TRUE.),'ALL 1-D & scalar, char, w/ NOT=T')
-  !ASSERT(.NOT. ALL(testString,testStringArray,.FALSE.),'ALL(testString,testStringArray,.FALSE.), bad')
-  ASSERT(.NOT. ALL(testStringArray,testString,.FALSE.),'ALL 1-D & scalar, str, w/ NOT=F')
-  !ASSERT(.NOT. ALL(char10,testStringArray,.FALSE.),'ALL(char10,testStringArray,.FALSE.), bad')
-  ASSERT(.NOT. ALL(testStringArray,char10,.FALSE.),'ALL 1-D & scalar, char, w/ NOT=F')
-  testString='test'
-  char10='test'
-  !ASSERT(ALL(testString,testStringArray),'ALL(testString,testStringArray)')
-  ASSERT(ALL(testStringArray,testString),'ALL 1-D & scalar, str')
-  !ASSERT(ALL(char10,testStringArray),'ALL(char10,testStringArray)')
-  ASSERT(ALL(testStringArray,char10),'ALL 1-D & scalar, char')
-  !ASSERT(.NOT.ALL(testString,testStringArray,.TRUE.),'ALL(testString,testStringArray,.TRUE)')
-  ASSERT(.NOT.ALL(testStringArray,testString,.TRUE.),'ALL 1-D & scalar, str, w/ NOT=T')
-  !ASSERT(.NOT.ALL(char10,testStringArray,.TRUE.),'ALL(char10,testStringArray,.TRUE)')
-  ASSERT(.NOT.ALL(testStringArray,char10,.TRUE.),'ALL 1-D & scalar, char, w/ NOT=T')
-  !ASSERT(ALL(testString,testStringArray,.FALSE.),'ALL(testString,testStringArray,.FALSE.)')
-  ASSERT(ALL(testStringArray,testString,.FALSE.),'ALL 1-D & scalar, str, w/ NOT=F')
-  !ASSERT(ALL(char10,testStringArray,.FALSE.),'ALL(char10,testStringArray,.FALSE.)')
-  ASSERT(ALL(testStringArray,char10,.FALSE.),'ALL 1-D & scalar, char, w/ NOT=F')
+!-------------------------------------------------------------------------------
+!> @brief 
```

Comment:
either take out the brief bits or fill them in...not really necessary for test routines

Reply:
deleted